client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -184,15 +213,16 @@ export default localize(
 	connect(
 		( state, ownProps ) => {
 			const verticals = getVerticals( state, ownProps.searchValue );
-			const isVerticalSearchPending = ownProps.searchValue && null == verticals;
 			return {
 				verticals: verticals || [],
-				isVerticalSearchPending,
 				defaultVertical: get( getVerticals( state, 'business' ), '0', {} ),
 			};
 		},
 		( dispatch, ownProps ) => ( {
+			bindDebounce: func => debounce( func, 1000 ),
 			shouldShowPopularTopics: searchValue => ! searchValue && ownProps.showPopular,
+			isVerticalSearchPending: ( searchValue, verticals = [] ) =>
+				searchValue && 0 === verticals.length,
```

Comment:
Hmmm, anything is possible. If it can't find a matching vertical, the API will always return one result, that is, whatever the user has inputted. 

But it won't hurt to be defensive here.

👍

Reply:
Now that I think of it, I think that was one of the reasons why I added `isFetching` to the state 🤣